src/pages/Top.js

```diff
@@ -1,18 +1,31 @@
-import React, { useState, useEffect } from 'react';
+import React, { useState, useEffect, useRef } from 'react';
 import Login from '../components/Login';
 import SignedIn from '../components/SignedIn';
 import ProtonSDK from '../utils/proton';
 
+const usePrevious = (value) => {
+  const ref = useRef();
+  useEffect(() => {
+    ref.current = value;
+  }, [value]);
+  return ref.current;
+}
+
 const Top = () => {
   const [error, setError] = useState('');
   const [auth, setAuth] = useState('');
   const [permission, setPermission] = useState('');
   const [accountData, setAccountData] = useState({});
 
+  const prevError = usePrevious(error);
   useEffect(() => {
```

Comment:
Sorry I missed this the first time - let's set up a separate `useEffect` hook for the `setError('')` to separate the error clearing from the session restoration.

https://reactjs.org/docs/hooks-effect.html#tip-use-multiple-effects-to-separate-concerns